streams/lms_metadata.py

```diff
@@ -0,0 +1,217 @@
+        track_data = self.flatten(track_load['result']['playlist_loop'])
+
+
+        if song_data.get('artist'):
+          self.meta.artist = song_data.get('artist') or "Loading..."
+          self.meta.album = song_data.get('album') or "Loading..."
+          self.meta.track = song_data.get('title') or "Loading..."
+        else:
+          self.meta.artist = song_data.get('title') or "Loading..."
+          self.meta.album = song_data.get('remote_title') or "Loading..."
+          self.meta.track = track_data.get('title') or "Loading..."
+
+
+        if song_data.get('artwork_url'):
+          self.meta.image_url = song_data.get('artwork_url')
+        elif song_data.get('coverid'):
+          self.meta.image_url = f"http://{self.server}:{self.port}/music/{song_data['coverid']}/cover.jpg?id={song_data['coverid']}"
+        else:
+          self.meta.image_url = 'static/imgs/lms.png'
+
+        try:
+          self.meta.save_file(self.player_name)
+        except:
+          pass
+
+        if self.debug:
+          self.meta.print_meta()
+          filename_prefix = str(self.player_name).replace(' ', '_')
+          with open(f"{filename_prefix}_track_raw.json", "w", encoding="UTF-8") as f:
+            json.dump(track_load, f, indent = 2)
+          with open(f"{filename_prefix}_song_raw.json", "w", encoding="UTF-8") as f:
+            json.dump(song_load, f, indent = 2)
+      except Exception as e:
+        logging.exception(f"Error: {e}, trying again in {self.meta_ref_rate} seconds...")
+
+      time.sleep(self.meta_ref_rate)
+
+if __name__ == '__main__':
+  # parser = argparse.ArgumentParser(description="LMS Metadata Reader - a script for finding an LMS server and extracting the name of the song, album, and artist as well as getting the album picture of the currently playing song")
+  # parser.add_argument('--name', type=str, required=True, help='The name of the LMS Player')
+  # parser.add_argument('--server', type=str, default=None, help='The hostname or IP of the computer running the LMS server', metavar="HOSTNAME")
+  # parser.add_argument('--port', type=int, default=9000, help='The port the LMS server uses')
+  # parser.add_argument('--ref', type=int, default=2, help='The frequency of metadata refresh cycles')
+  # parser.add_argument('--debug', action='store_true', default=False, help='''debug mode, activates various console logs so that you can debug in the command line,
+  #                     also creates json dumps in the main directory: {player name}_track_raw.json and {player name}_song_raw.json''')
+  # args = parser.parse_args()
+
+  # There's a few args here that aren't currently used by the system, but can be helpful for debugging in the field
+  # some are also there so we can implement future features easier, or so the devs at home can do the same with a minimal lift
+  # LMSMetadataReader(args.name, args.server, args.port, args.ref, args.debug).connect()
+  LMSMetadataReader("LinkData", None, 9000, 2, True).connect()
```

Comment:
Left local test changes in on accident, these will be cleaned out in next commit